src/backend/distributed/utils/shard_utils.c

```diff
@@ -0,0 +1,31 @@
+/*-------------------------------------------------------------------------
+ *
+ * shard_utils.c
+ *
+ * This file contains functions to perform useful operations on shards.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "utils/lsyscache.h"
+
+#include "distributed/relay_utility.h"
+#include "distributed/shard_utils.h"
+
+/*
+ * GetShardOid returns the oid of the shard from the given distributed relation
+ * with the shardid.
+ */
+Oid
+GetShardOid(Oid distRelId, uint64 shardId)
```

Comment:
I find this API dangerous. This only works for local shards. Can you maybe rename the function to `GetLocalShardId` or such?

Reply:
https://github.com/citusdata/citus/pull/3557/files#diff-04ee83e6d1c2aebae3aad31b888ce1a4R23